### PR TITLE
[15.0][ADD] sale_order_whole_delivered_invoiceability: Add new module

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT50680

@pedrobaeza @chienandalu please review

### CODE REVIEW

File: sale_order_whole_delivered_invoiceability/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Create a product and set as invoicing policy ‘Delivered Quantities’.
```
Comment (pedrobaeza) [inline sale_order_whole_delivered_invoiceability/readme/CONFIGURE.rst]:
```suggestion
#. Create a new storable product
#. Set its invoicing policy as "Delivered Quantities".
```

File: sale_order_whole_delivered_invoiceability/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    )
+
+    @api.depends("partner_id")
+    def _compute_whole_delivered_invoiceability(self):
+        for record in self:
+            if record.partner_id.whole_delivered_invoiceability:
+                record.whole_delivered_invoiceability = True
+
+    @api.depends("whole_delivered_invoiceability")
+    def _get_invoice_status(self):
```
Comment (pedrobaeza) [inline sale_order_whole_delivered_invoiceability/models/sale_order.py:23]: Remember to put proper comments acting as docstrings

```suggestion
    def _get_invoice_status(self):
        # Intercept the invoice_status computed method to
        # set it as not invoiceable if the delivered quantity
        # is less than the ordered quantity.
```

File: sale_order_whole_delivered_invoiceability/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+This module controls the invoicing of sale orders until all material has been delivered.
```
Comment (pedrobaeza) [inline sale_order_whole_delivered_invoiceability/readme/DESCRIPTION.rst]:
```suggestion
This module prevents the invoiceability of the sale orders until all the
stockable products has been delivered.
```

Comment (pilarvargas-tecnativa) [reply sale_order_whole_delivered_invoiceability/readme/DESCRIPTION.rst]: All changes done! thanks!

File: sale_order_whole_delivered_invoiceability/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        # Intercept the invoice_status computed method to
+        # set it as not invoiceable if the delivered quantity
+        # is less than the ordered quantity.
+        res = super()._get_invoice_status()
+        for order in self:
+            if order.whole_delivered_invoiceability:
```
Comment (chienandalu) [inline sale_order_whole_delivered_invoiceability/models/sale_order.py]: You can save on level here
```suggestion
        for order in self.filtered("whole_delivered_invoiceability"):
```

File: sale_order_whole_delivered_invoiceability/models/sale_order.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                lines_to_invoice = order.order_line.filtered(
+                    lambda line: line.product_id.invoice_policy == "delivery"
+                    and line.product_uom_qty > line.qty_delivered
```
Comment (chienandalu) [inline sale_order_whole_delivered_invoiceability/models/sale_order.py]: You should use `float_compare` here

File: sale_order_whole_delivered_invoiceability/models/sale_order.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    @api.depends("whole_delivered_invoiceability")
+    def _get_invoice_status(self):
+        # Intercept the invoice_status computed method to
+        # set it as not invoiceable if the delivered quantity
+        # is less than the ordered quantity.
+        res = super()._get_invoice_status()
+        for order in self:
+            if order.whole_delivered_invoiceability:
+                lines_to_invoice = order.order_line.filtered(
```
Comment (chienandalu) [inline sale_order_whole_delivered_invoiceability/models/sale_order.py]: This name is a little bit missleading, as we want to set lines that aren't invoiceable
```suggestion
                uncomplete_lines = order.order_line.filtered(
```

**Review by chienandalu — COMMENTED** [review]
Thanks :) Some code review:

File: sale_order_whole_delivered_invoiceability/models/sale_order.py (hunk 5 above)
Comment (pilarvargas-tecnativa) [reply sale_order_whole_delivered_invoiceability/models/sale_order.py]: done! thanks!

**Review by chienandalu — APPROVED** [review]
Thanks! :+1:

**Pull request merged**